analysis.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2023 IBM
+# Author: Misbah Anjum N <misanjum@linux.vnet.ibm.com>
+
+
+'''
+    analysis.py script is aimed to help in analysis/comparison of avocado test runs 
+    by generating a simple excel file (.xlsx). The results.json file which gets created 
+    after avocado run is passed as input in command line while running this script and 
+    depending on the flag/options provided, the excel analysis/omparison sheet will be 
+    generated.
+
+    Prerequsites:-
+    pip3 install pandas[excel]
+
+    flags/options:-
+    1. --new-analysis
+    2. --add-to-existing
+    3. --compare-two-results
+
+    python3 analysis.py --new-analysis <json_file>
+    python3 analysis.py --add-to-existing <xlsx_file> <json_file>
+    python3 analysis.py --compare-two-results <old_json_file> <new_json_file>
+
+    Check README.md for more explanation
+'''
+
+
+import sys
+import json
+import pandas as pd
+from openpyxl.styles import Font, Border, Side, PatternFill, Alignment
+
+
+def new_analysis(data):
+    '''
+        This function is used to generate an excel file which contains analysis having 
+        test case name as "ID", test status as "Status" and fail reason if any as "Fail 
+        Reason". It can used to generate analysis on a single test run results.json file.
```

Comment:
seems require re-wording , can you please take look

Reply:
Reworded the function descriptions, please have a look.
Thanks